include/cuco/detail/hash_functions/identityhash.cuh

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuco/detail/hash_functions/utils.cuh>
+
+namespace cuco::detail {
+
+/**
+ * @brief An Identity hash function to hash the given argument on host and device
+.*
+ * @note IdentityHash is perfect iff hash_table_capacity >= |input set|
```

Comment:
```suggestion
 * @note `identity_hash` is perfect iff `hash_table_capacity >= |input set|`
```

nit: Doxygen can render inline code which makes it a bit easier to parse.

Reply:
Good catch! Inline code is certainly easier to parse